### DISCUSSION TITLE
Module local variable

### DISCUSSION THREAD

**Discussion by gsuuon** [nushell/nushell#7067] [Q&A] [answered]
How do I write a module scoped variable? 

I'd like to write something like this:
```nu
# scripts/meet.nu
let greeting = "hi"
export def greet [name: string] {
    $"($greeting) ($name)"
}
```
Without `greeting` polluting the parent scope.

The obvious way errors with `expected def or export keyword`. I could also define a command returning a constant. What's the right way to do this?

**Comment by kubouch** [comment, marked as answer]
It's not possible to have module-local variables yet. We plan to investigate if it would be possible but for now you need to wrap it in a custom command.

**Reply by solodov** [reply]
is there a feature request I can cc myself on? I ran into this limitation on the second day of using nushell, I suspect it's a pretty common thing.

**Reply by kubouch** [reply]
I don't think there is an issue for it. You can make one if you don't find any. I'm planning to investigate it eventually. We likely won't be able to support storing everything into the variable but we could have something similar to `const` in Rust that would allow to store things like static strings / integers etc.

**Reply by solodov** [reply]
filed https://github.com/nushell/nushell/issues/7416
